File: 2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.java
class Solution {
    int countSetBits(int n)
    {
        int count = 0;
        while (n > 0) {
            n &= (n - 1);
            count++;
        }
        return count;
    }
    public int minOperations(int[] nums, int k) {
        int all = 0;
        for(int i : nums) all ^= i;
        int t = all^k;
        return countSetBits(t);
    }
}

File: 2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        
        def count(n: int) -> int:
            count = 0
            while n>0:
                n = n&(n-1)
                count += 1
            return count
        
        xor = 0
        for i in nums: xor ^= i
        return count(xor^k)